pandas/tseries/tests/test_tslib.py

```diff
@@ -608,6 +608,19 @@ def test_parsers(self):
         self.assertTrue(result1 is tslib.NaT)
         self.assertTrue(result1 is tslib.NaT)
 
+    def test_datetime_bool(self):
+        # GH13176
+        self.assertEqual(to_datetime(False), to_datetime(0))
```

Comment:
can you also add some tests where bools are intermixed with valid (and also invalid conversions)